core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -345,6 +349,13 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
           createInMemoryStore(attempt)
       }
     } catch {
+      case _: FileNotFoundException if this.conf.get(EVENT_LOG_ROLLING_ON_DEMAND_LOAD_ENABLED) =>
+        if (app.attempts.head.info.appSparkVersion == "unknown") {
+          listing.synchronized {
+            listing.delete(classOf[ApplicationInfoWrapper], appId)
+          }
+        }
+        return None
```

Comment:
Hmm, I don't quite understand what this did. Seems `loadFromFallbackLocation` loads a dummy record from the log path into `listing`?

But what does this `FileNotFoundException` catch? Why it deletes the dummy record immediately?

Reply:
Thank you for review, @viirya .

Yes, this adds a dummy record (based on the user request) to proceed to load the actual file. However, if the actual file doesn't exist, `FoundNotFoundException` will be thrown. It means it's a user mistake. In that case, since we don't need this dummy record, we cleaned up.